lib/elixir/lib/kernel/typespec.ex

```diff
@@ -1216,4 +1216,27 @@ defmodule Kernel.Typespec do
   defp unpack_typespec_kw(_, _acc) do
     :error
   end
+
+  # Called by elixir_erl.
+  def spec_related_info(table) do
+    translate_type = fn {kind, expr, caller} -> translate_type(kind, expr, caller) end
+    translate_spec = fn {kind, expr, caller} -> translate_spec(kind, expr, caller) end
```

Comment:
Isn't this just &translate_type/3 and &translate_spec/3?

Reply:
We are passing three-element tuples around, not three arguments :)